### PR TITLE
Deprecate quarkus-bom-deployment and merge its content into quarkus-bom

### PR DESCRIPTION
**Pull request by aloubyansky**
First step towards fixing #9490

### CODE REVIEW

File: extensions/smallrye-openapi-common/deployment/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-metadata:
-  keywords:
-  - "smallrye-openapi"
-  - "openapi"
-  - "open-api"
-  guide: "https://quarkus.io/guides/openapi-swaggerui"
-  categories:
-  - "web"
-  status: "stable"
-  unlisted: true
```
Comment (gsmet) [inline extensions/smallrye-openapi-common/deployment/src/main/resources/META-INF/quarkus-extension.yaml:12]: Why did you delete this file?

File: build-parent/pom.xml [hunk 2]
```diff
@@ -209,6 +209,34 @@
...
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-test-maven</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-security-test-utils</artifactId>
+                <version>${project.version}</version>
+                <scope>test</scope>
+            </dependency>
```
Comment (gsmet) [inline build-parent/pom.xml:239]: Why do we have to inject these dependencies here?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added two small questions, we need to figure out at least the one about the extension descriptor.

You don't need any code adjustments?

File: extensions/smallrye-openapi-common/deployment/src/main/resources/META-INF/quarkus-extension.yaml (hunk 1 above)
Comment (aloubyansky) [reply extensions/smallrye-openapi-common/deployment/src/main/resources/META-INF/quarkus-extension.yaml:12]: This file does not belong in the deployment artifact. And there is no corresponding runtime one.

File: build-parent/pom.xml (hunk 2 above)
Comment (aloubyansky) [reply build-parent/pom.xml:239]: They were in the deployment bom, they don't belong in the runtime one, so I moved them to build-parent.

**Review by gsmet — APPROVED** [review]
